src/crafting.rs

```diff
@@ -231,13 +237,37 @@ pub fn calculate_crafting_profit(
     let mut total_crafting_cost = Rational32::zero();
     let mut crafting_count = 0;
     let mut total_crafting_steps = Rational32::zero();
+    let mut used_recipes = HashMap::new();
```

Comment:
Could `used_recipes` be a `HashSet`?

Reply:
I chose a HashMap because I wanted to track used recipes which are known and unknown. You are right though that I ultimately only care about unknown recipes - and I know which are which when I insert them. Good change.